public/app/features/profile/ChangePasswordForm.tsx

```diff
@@ -1,20 +1,27 @@
 import React, { FC } from 'react';
 import config from 'app/core/config';
+import { UserDTO } from 'app/types';
 import { Button, LinkButton, Form, Field, Input, HorizontalGroup } from '@grafana/ui';
 import { ChangePasswordFields } from 'app/core/utils/UserProvider';
 import { css } from 'emotion';
 
 export interface Props {
+  user: UserDTO;
   isSaving: boolean;
   onChangePassword: (payload: ChangePasswordFields) => void;
 }
 
-export const ChangePasswordForm: FC<Props> = ({ onChangePassword, isSaving }) => {
+export const ChangePasswordForm: FC<Props> = ({ user, onChangePassword, isSaving }) => {
   const { ldapEnabled, authProxyEnabled } = config;
+  const authSource = user.authLabels?.length && user.authLabels[0];
 
   if (ldapEnabled || authProxyEnabled) {
     return <p>You cannot change password when ldap or auth proxy authentication is enabled.</p>;
   }
+  if (authSource) {
+    return <p>Password cannot be changed here!</p>;
```

Comment:
Can we remove this? We have no way currently to remove auth sources, so with this change there would be no way to go back to user name + password after disabling the oauth.

Reply:
I guess we just need a better if then?
Cause the thing is we want to hide change password in Grafana for users logged in via OAuth. As it will just give confusion if they are able to change the password there, but it will not do anything..

Reply:
but the condition needs to be linked to disableLoginForm, not just the authSource